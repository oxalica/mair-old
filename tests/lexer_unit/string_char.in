"\"x'"
"abc劲"
"\x7aa"
"\u{1}"
"\u{2764}"
"\\\n\r a"
"\\\'\"\n\r\t\0"
r"\"
r"\n"
r"xx\u{}"
br"\"
'劲'
b'\x00'
'\''
'\\'
'\u{99}'
'\u{000000}'
'\u{999999}'
'\t'
'\a'
'\x0'
'\x0*'
"aa\u{}"
''
"
