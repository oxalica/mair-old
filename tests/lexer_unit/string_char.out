Ok([(Literal(StrLike { is_bytestr: false, s: "\"x\'" }), "\"\\\"x\'\"")])
Ok([(Literal(StrLike { is_bytestr: false, s: "abc劲" }), "\"abc劲\"")])
Ok([(Literal(StrLike { is_bytestr: false, s: "za" }), "\"\\x7aa\"")])
Ok([(Literal(StrLike { is_bytestr: false, s: "\u{1}" }), "\"\\u{1}\"")])
Ok([(Literal(StrLike { is_bytestr: false, s: "❤" }), "\"\\u{2764}\"")])
Ok([(Literal(StrLike { is_bytestr: false, s: "\\\n\r a" }), "\"\\\\\\n\\r a\"")])
Ok([(Literal(StrLike { is_bytestr: false, s: "\\\'\"\n\r\t\u{0}" }), "\"\\\\\\\'\\\"\\n\\r\\t\\0\"")])
Ok([(Literal(StrLike { is_bytestr: false, s: "\\" }), "r\"\\\"")])
Ok([(Literal(StrLike { is_bytestr: false, s: "\\n" }), "r\"\\n\"")])
Ok([(Literal(StrLike { is_bytestr: false, s: "xx\\u{}" }), "r\"xx\\u{}\"")])
Ok([(Literal(StrLike { is_bytestr: true, s: "\\" }), "br\"\\\"")])
Ok([(Literal(CharLike { is_byte: false, ch: '劲' }), "\'劲\'")])
Ok([(Literal(CharLike { is_byte: true, ch: '\u{0}' }), "b\'\\x00\'")])
Ok([(Literal(CharLike { is_byte: false, ch: '\'' }), "\'\\\'\'")])
Ok([(Literal(CharLike { is_byte: false, ch: '\\' }), "\'\\\\\'")])
Ok([(Literal(CharLike { is_byte: false, ch: '\u{99}' }), "\'\\u{99}\'")])
Ok([(Literal(CharLike { is_byte: false, ch: '\u{0}' }), "\'\\u{000000}\'")])
Err(LexicalError { loc: "\'\\u{999999}\'", kind: InvalidEscape })
Ok([(Literal(CharLike { is_byte: false, ch: '\t' }), "\'\\t\'")])
Err(LexicalError { loc: "\'\\a\'", kind: UnknowToken })
Err(LexicalError { loc: "\'\\x0\'", kind: UnknowToken })
Err(LexicalError { loc: "\'\\x0*\'", kind: UnknowToken })
Err(LexicalError { loc: "\"aa\\u{}\"", kind: InvalidEscape })
Err(LexicalError { loc: "\'\'", kind: UnknowToken })
Err(LexicalError { loc: "\"", kind: UnterminatedString })
